src/Widgets/AbstractAppContainer.vala

```diff
@@ -80,6 +82,19 @@ namespace AppCenter {
             }
         }
 
+        private string? suggested_amount_ = null;
+        public string suggested_amount {
+            get {
+                if (this.package == null || this.package.component == null || updates_view) {
+                    return DEFAULT_PRICE_DOLLARS;
+                } else if (suggested_amount_ == null) {
+                    suggested_amount_ = this.package.component.get_custom_value ("x-appcenter-suggested-price");
+                }
+
+                return suggested_amount_ != null ? suggested_amount_ : DEFAULT_PRICE_DOLLARS;
+            }
+        }
+
```

Comment:
Thanks @Philip-Scott , I agree about setting the price to 0 if no stripe key is found, I'll do that later. However, I think we should keep this code out of HumbleButton.vala . If we do that, it will depend on a Package object, while at the moment it's a generic class that just expects an amount. I think it's probably better to keep it generic, so it can be used in other cases if needed. We'll see what other people think.